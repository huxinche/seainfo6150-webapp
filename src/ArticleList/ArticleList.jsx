import React from "react";
import PropTypes from "prop-types";
import ArticleListItem from "../ArticleListItem/ArticleListItem.jsx";
import styles from './ArticleList.module.css';

const ArticleList = (props) => {
  return (
    <div className={styles.list}>
      {props.articles.map((article) => (
        <ArticleListItem article={article} key={article.slug} slug={article.slug}/>
      ))}
    </div>
  );
};

ArticleList.propTypes = {
  articles: PropTypes.array.isRequired
}
export default ArticleList;
