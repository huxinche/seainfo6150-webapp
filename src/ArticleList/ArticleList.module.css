.list {
    border-radius: 6px;
    border: 1px dashed gray;
    background: #bbb;
    box-shadow: 0 0 10px rgba(200, 0, 0, 0.6);
    margin: 16px;
    display: grid;
}

@media screen and (max-width: 600px) {
    .list {
        grid-template-columns: 1fr;
        margin: 8px 24px;
    }

    .list p {
        font-size: 15px;
    }
}

@media (max-width: 900px) and (min-width: 600px) {
    .list {
        grid-template-columns: 1fr 1fr;
    }
    .list p {
        font-size: 15px;
    }

    .list > div {
        margin: 8px;
    }
}

@media (max-width: 1400px) and (min-width: 900px) {
    .list {
        grid-template-columns: 1fr 1fr 1fr;
        margin: 24px;
    }
    .list > div {
        margin: 8px;
    }
}

@media screen and (min-width: 1400px) {
    .list {
        grid-template-columns: 1fr 1fr 1fr 1fr;
        margin: 30px;
        border-radius: 8px;
        background: #eee;
    }
    .list > div {
        margin: 16px 14px;
    }
}
