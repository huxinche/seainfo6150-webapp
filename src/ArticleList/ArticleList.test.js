import React from "react";
import ArticleList from "./ArticleList";
import article from '../testData.json'

describe("ArticleList tests", () => {
  it("renders correctly", () => {
    const { container } = render(<ArticleList articles={[article]} />);
    expect(container).toMatchSnapshot();
  });
});
