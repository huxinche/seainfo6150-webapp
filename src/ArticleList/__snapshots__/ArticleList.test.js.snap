// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleList tests renders correctly 1`] = `
<div>
  <div
    class="list"
  >
    <div
      class="card"
    >
      <div
        class="main"
      >
        <img
          alt="You can take a pie on a plane"
          class="image"
          src="https://cdn.cnn.com/cnnnext/dam/assets/170321135754-airport-security-line-file-super-169.jpg"
        />
        <a
          class="title"
          href="/articlelist/170321135754-airport-security-line"
        >
          You can take a pie on a plane
        </a>
      </div>
      <div
        class="info"
      />
      <button
        class="button"
      >
        Show more
      </button>
    </div>
  </div>
</div>
`;
